test/common/http/http2/codec_impl_test.cc

```diff
@@ -1749,7 +1749,12 @@ TEST_P(Http2CodecImplTest, PingFlood) {
         buffer.move(frame);
       }));
 
-  EXPECT_THROW(client_->sendPendingFrames(), ServerCodecError);
+  // Legacy codec does not propagate error details and uses generic error message
+  EXPECT_THROW_WITH_MESSAGE(
```

Comment:
nit: it seems odd to me that in the tests we still use exceptions. Would these be better written as a helper which pivots what it does based on new vs. old and then we can just clean up old later? I don't feel strongly about it.

Reply:
This is the artifact of test construction as it uses codec to test codec. So when dispatch of the codec under test fails the stack has to be unwound through the client codec that was used to build the test wire bytes. The test really needs to be redesigned to avoid doing this. Note that these exceptions are specific to the test, they are not codec exceptions.